include/dos_system.h

```diff
@@ -196,11 +204,13 @@ class DOS_Drive_Cache {
 	};
 
 private:
+	DOS_Drive_Cache(const DOS_Drive_Cache&); // prevent copying
+	DOS_Drive_Cache& operator= (const DOS_Drive_Cache&); // prevent assignment
```

Comment:
With C++11 we can make it even more explicit:

    DOS_Drive_Cache(const DOS_Drive_Cache&) = delete;

This has added benefit of not touching the vtable of original object - compiler knows, that this constructor is not legal and that's it (it does not need to be private).

(I don't remember if it's possible to mark assignment operator as deleted).